packages/components/molecules/f-restaurant-card/src/components/subcomponents/_tests/IconText.test.js

```diff
@@ -0,0 +1,69 @@
+import { shallowMount, config } from '@vue/test-utils';
+import sut from '../IconText.vue';
+
+describe('IconText.vue', () => {
+    it('displays the provided text', () => {
+        // arrange
+        const propsData = {
+            text: 'foo'
+        };
+
+        // act
+        const wrapper = shallowMount(sut, { propsData });
+
+        const visibleText = wrapper.find('[data-test-id="icon-text-visible-text"]').text();
+
+        // assert
+        expect(visibleText).toBe('foo');
+    });
+
+    describe('Icon Visibility', () => {
+        describe('Tile view', () => {
+            it('should hide icon when "hideIconInTileView" is true', () => {
+                // arrange
+                const propsData = {
+                    text: 'foo',
+                    hideIconInTileView: true
+                };
+
+                const slots = { default: '<div></div>' };
+
+                // act
+                const wrapper = shallowMount(sut, { propsData, slots });
+
+                const iconExists = wrapper.find('[data-test-id="icon-text-icon"]').exists();
+
+                // assert
+                expect(iconExists).toBe(false);
+            });
+        });
+
+        describe('List view', () => {
+            beforeAll(() => {
+                config.provide = { isListItem: true };
+            });
+
+            afterAll(() => {
+                config.provide = {};
+            });
```

Comment:
I think we might prefer beforeEach and afterEach here as this sounds like it would preserve state between future tests added (might be wrong here)

Reply:
This sets up the component in the list item view, which is what all the tests in the describe block are testing in, no other state would be preserved between tests